lib/elixir/lib/module/checker.ex

```diff
@@ -1,16 +1,212 @@
 defmodule Module.Checker do
-  def verify(module_map, _binary) do
+  @moduledoc false
+
+  defmodule Coordinator do
```

Comment:
I think the coordinator should be the process spawning processes (aka the parallel checker). In the future, the coordinator needs to know exactly what module to run and not run in order to solve dependencies. For example, if A depends on B, the coordinator may prioritize running B.

Reply:
I will leave a TODO for this, right now it only adds complexity.

Reply:
I think it would actually make things simpler because we can use a GenServer here, which is better than the unstructured code in the parallel compiler. :D

Reply:
Okay, I will try it.